### PR TITLE
Changes to how the button handles colors. Demo project added. 

### PR DESCRIPTION
**Pull request by andrewlloyd100**
Now if button title color is updated once spinner is loading the title will be shown with the new color. When the spinner finishes loading the colors will only be restored if they haven't been changed since showing the spinner. Demo Project added to Repo.

### CODE REVIEW

File: Spinner/Spinner.swift [hunk 1]
```diff
@@ -84,25 +84,23 @@ public class SpinnerView: NSObject, Spinner {
         let view = showSpinner(inView: button, style: style, color: color)
         
         if let spinnerView = view as? SpinnerView {
-            spinnerView.userInteractionEnabledAtReception = button.isUserInteractionEnabled
```
Comment (dominik-hadl) [inline Spinner/Spinner.swift:87]: Why was this deleted? :)

Comment (andrewlloyd100) [reply Spinner/Spinner.swift:87]: Its setting it twice, it sets it when you set the spinner in the showSpinner(inView:_ call. Because of this, it was causing the buttons user interaction to be disabled after the spinner was dismissed even though when the spinner was first shown, interaction was enabled.

Comment (dominik-hadl) [reply Spinner/Spinner.swift:87]: @andrewlloyd100  Ah, I see. Ok then :) Can you verify this thing also works for custom spinner?

Comment (andrewlloyd100) [reply Spinner/Spinner.swift:87]: Sure, I haven't added a custom spinner to the Demo yet actually so I'll do that now

Comment (mariusc) [reply Spinner/Spinner.swift:87]: Ah, so that's where the problem came from. I thought I noticed yesterday that the button is still deactivated after the spinner is dismissed, but I didn't have the time to investigate too much. Thanks for solving this :D. We also need to update the Swift 2.3 branch with this fix now. Would be nice to have a test that catches this situation :D.

**Review by dominik-hadl — APPROVED** [review]
